accord-core/src/main/java/accord/impl/InMemoryCommand.java

```diff
@@ -218,21 +230,23 @@ public Status status()
     }
 
     @Override
-    public void status(Status status)
+    public void setStatus(Status status)
     {
         this.status = status;
     }
 
     @Override
-    public boolean isGloballyPersistent()
+    public Durability durability()
     {
-        return isGloballyPersistent;
+        if (status.hasBeen(PreApplied) && durability == NotDurable)
```

Comment:
Why didn't we `setDurability` to local when transitioning to `PreApplied`?

Reply:
We could do, but it's also an unnecessary additional step, since it's defined by the `Status`. Not sure it's clearer to propagate it when updating the status than to use the status as a lower bound when reading the `durability`.

In part I assume that `status` will always be in memory, and in part I think it is an easier place to impose the semantic. But I don't feel very strongly about it.

Reply:
`AccordCommand` doesn't also do this check? Should it not be in the integration so it can be common across command implementations?

Reply:
Should be fixed